src/smtp.rs

```diff
@@ -19,8 +26,15 @@ pub fn send(account: &Account, msg: &lettre::Message) -> Result<()> {
         SmtpTransport::relay
     };
 
+    let tls = TlsParameters::builder(account.smtp_host.to_string())
+        .dangerous_accept_invalid_hostnames(account.smtp_insecure())
+        .dangerous_accept_invalid_certs(account.smtp_insecure())
+        .build()
+        .unwrap();
+
     smtp_relay(&account.smtp_host)?
         .port(account.smtp_port)
+        .tls(Tls::Wrapper(tls))
```

Comment:
I just saw this PR, but forcing `Wrapper` means that even when starttls is turned on, TLS will be used anyway

Reply:
But `STARTTLS` is just a `TLS` extension, so theoretically the `.tls(...)` method may affect only the `TLS` part without interfering with the `STARTTLS` part? I guess integration tests need to be set up to catch all those tricky behaviours.

Reply:
The [`relay`](https://docs.rs/lettre/0.10.0-beta.3/src/lettre/transport/smtp/transport.rs.html#51-57) and [`starttls_relay`](https://docs.rs/lettre/0.10.0-beta.3/src/lettre/transport/smtp/transport.rs.html#71-77) methods are just convenience functions for setting the right variant of the [`Tls`](https://docs.rs/lettre/0.10.0-beta.3/lettre/transport/smtp/client/enum.Tls.html) enum.

`Tls::Required` = The server either supports `STARTTLS` and allows lettre to upgrade the initial plain text connection to TLS or we fail.
`Tls::Wrapped` = We open the connection directly with TLS.

Reply:
I'm a Rust/TLS noob but I did not see any issue with starttls on the few tests I drove.

Reply:
I didn't realized you are part of the lettre core team @paolobarbolini! Thanks for your precious help. So it means that the `insecure` options are not compatible with `STARTTLS`? I mean, there is no way to enable `STARTTLS` AND pass a custom TLS builder with `dangerous_accept_*` options?

Reply:
> So it means that there is no way to enable `STARTTLS` AND pass a custom TLS builder with `dangerous_accept_*` options?

It's totally possible. The change that needs to be done is passing `.tls(...)` the `Tls::Required(tls)` variant instead of the `Tls::Wrapper(tls)` variant when `account.smtp_starttls()` is `true`.

Reply:
Awesome! @remche would you like to take care of this?

Reply:
Will do asap